pallets/rewards/src/lib.rs

```diff
@@ -427,4 +427,17 @@ pub mod pallet {
 			Currencies::<T, I>::get(currency_id).0
 		}
 	}
+
+	impl<T: Config<I>, I: 'static> Pallet<T, I>
+	where
+		RewardAccountOf<T, I>: FullCodec + Default,
+	{
+		pub fn list_currencies(
+			account_id: T::AccountId,
+		) -> sp_std::vec::Vec<(T::DomainId, T::CurrencyId)> {
```

Comment:
Maybe we can avoid the `Vec` allocation returning directly the iterator. Something like: `impl Iterator<Item = (T::DomainId, T::CurrencyId)>` in the result type.

Reply:
Not sure if this will complicate things a bit when implementing it in the runtime APIs and if this optimization is worth it.

Reply:
Its ok for me leave the `Vec` if the things get complicated dealing the Iterator trait 👍🏻